Mathlib/GroupTheory/Subgroup/Basic.lean

```diff
@@ -3498,6 +3508,20 @@ instance (priority := 100) Subgroup.normal_subgroupOf {H N : Subgroup G} [N.Norm
 #align subgroup.normal_subgroup_of Subgroup.normal_subgroupOf
 #align add_subgroup.normal_add_subgroup_of AddSubgroup.normal_addSubgroupOf
 
+theorem Subgroup.map_normalClosure_equiv_eq_normalClosure_image {s : Set G} (f : G ≃* N) :
+    (normalClosure s).map f = normalClosure (f '' s) := by
+  have : Normal (map f (normalClosure s)) := Normal.map inferInstance f f.surjective
```

Comment:
I pushed a golf here: the result is that this lemma should only need `f` to be surjective, not an equivalence. Can you update things accordingly?

Reply:
I think I originally misunderstood what you meant here and pushed still using `f` as an equivalence.  Will take another stab at it.

Reply:
What you pushed is what *I* originally meant, but @alreadydone suggested it might be true more generally

Reply:
I was able to prove the lemma that @alreadydone mentioned but not really sure it fits well into the other proofs.  I also generalized to use only the surjection for what you showed above but haven't found out why I can get `map_normalClosure` to work but not `map_normalClosure'`.  At any rate, just let me know what direction you think is best for these lemmas and I will try wrapping it all up in that direction:

```
theorem Subgroup.map_normalClosure {s : Set G} (f : G ≃* N) :
    (normalClosure s).map f = normalClosure (f '' s) := by
  have : Normal (map f (normalClosure s)) := Normal.map inferInstance f f.surjective
  apply le_antisymm
  · rw [map_le_iff_le_comap]
    refine normalClosure_le_normal ?_
    simp only [coe_comap, ← Set.image_subset_iff]
    exact subset_normalClosure
  · exact normalClosure_le_normal (Set.image_subset f subset_normalClosure)

theorem Subgroup.map_normalClosure' {s : Set G} (f : G →* N) (h : Function.Surjective f) :
    (normalClosure s).map f = normalClosure (f '' s) := by
  have : Normal (map f (normalClosure s)) := Normal.map inferInstance f h
  apply le_antisymm
  · rw [map_le_iff_le_comap]
    refine normalClosure_le_normal ?_
    simp only [coe_comap, ← Set.image_subset_iff]
    exact subset_normalClosure
  · exact normalClosure_le_normal (Set.image_subset f subset_normalClosure)

theorem Subgroup.normalClosure_preimage_le_comap {s : Set N} (f : G →* N) :
    normalClosure (f ⁻¹' s) ≤ (normalClosure s).comap f := by
  refine normalClosure_le_normal ?h
  simp only [coe_comap]
  refine Set.preimage_mono ?h.h
  exact subset_normalClosure

theorem Subgroup.comap_normalClosure {s : Set N} (f : G ≃* N) :
    normalClosure (f ⁻¹' s) = (normalClosure s).comap f  := by
  have := Set.preimage_equiv_eq_image_symm s f.toEquiv
  simp_all [comap_equiv_eq_map_symm, map_normalClosure]

theorem Subgroup.comap_normalClosure' {s : Set N} (f : G ≃* N) :
    normalClosure (f ⁻¹' s) = (normalClosure s).comap f  := by
  have := Set.preimage_equiv_eq_image_symm s f.toEquiv
  simp_all [comap_equiv_eq_map_symm]
  -- simp [@map_normalClosure' G _ N _ (f ⁻¹' s) f.toMonoidHom f.surjective]
```

Reply:
I think my comment above still applies; instead of using `f.surjective`, can you add `hf : Surjective f`, and replace `f` with a `MonoidHom`?

Reply:
Yes, update uses a `MonoidHom` and its surjection explicitly now.